README.md

```diff
@@ -1,7 +1,125 @@
-# TimeKeeper
+# Chorus TimeKeeper
 
-A super simple class for DI with timekeeping.
+The **Chorus TimeKeeper** is an elegant utility for Dependency Injection (DI) with timekeeping. At its core, it offers a straightforward approach to time management in applications, allowing for consistent time retrieval and even time manipulation for testing purposes. This can be exceptionally helpful when writing tests or simulating various time conditions in your code.
 
-```
+It provides two primary components:
+1. `TimeKeeper`: Fetches real-time data.
+2. `FakeTimeKeeper`: Helps in mocking or manipulating time during testing.
+
+For example, when testing time-sensitive components, you can utilize `FakeTimeKeeper` to preset time scenarios, avoiding the hassle of waiting in real-time or simulating date changes on your system.
+
+## Requirements
+
+- PHP ^7.4 or ^8.0
+
+## Installation
+
+You can install the package via Composer:
+
+```bash
 composer require tumblr/chorus-timekeeper
 ```
+
+
+## Basic Usage
+
+```php
+// Using the real TimeKeeper
+$timekeeper = new Tumblr\Chorus\TimeKeeper();
+
+echo $timekeeper->getCurrentUnixTime();  // Outputs the current unix epoch time.
+
+// Using the FakeTimeKeeper
+$fakeTimeKeeper = new Tumblr\Chorus\FakeTimeKeeper(1628700000);  // Set a specific unix time.
+
+echo $fakeTimeKeeper->getCurrentUnixTime();  // Outputs 1628700000.
+
+```
+
+## Development Workflow
+
+The project has a Makefile to simplify development processes. You can use various make commands to run tests, validate code standards, and more. Here's a quick overview:
+
+### 1. Prerequisites Setup
```

Comment:
I would usually say: call `make` and it does all you need. What's your take?

Reply:
It sounds good, but I generally like to make it as descriptive as possible. I usually keep that mentality: if the reader has never heard of the Makefile, how can I give them a hint?

I’ll certainly reconsider the balance between been descriptive and been redundant.